tools/integration-tests/tester/tests/testutil.go

```diff
@@ -258,13 +260,16 @@ func SendIotaTransaction(t *testing.T, from *framework.Peer, to *framework.Peer,
 		outputs = append(outputs, output)
 	}
 	txEssence := ledgerstate.NewTransactionEssence(0, time.Now(), txConfig.AccessManaPledgeID, txConfig.ConsensusManaPledgeID, ledgerstate.NewInputs(input), ledgerstate.NewOutputs(outputs...))
-	sig := ledgerstate.NewED25519Signature(from.KeyPair(0).PublicKey, from.KeyPair(0).PrivateKey.Sign(txEssence.Bytes()))
+	sig := ledgerstate.NewED25519Signature(from.KeyPair(txConfig.FromAddressIndex).PublicKey, from.KeyPair(txConfig.FromAddressIndex).PrivateKey.Sign(txEssence.Bytes()))
 	unlockBlock := ledgerstate.NewSignatureUnlockBlock(sig)
 	txn := ledgerstate.NewTransaction(txEssence, ledgerstate.UnlockBlocks{unlockBlock})
 
 	// send transaction
 	txId, err = from.SendTransaction(txn.Bytes())
-	require.NoErrorf(t, err, "could not send transaction on %s", from.String())
+	if err != nil {
```

Comment:
why change the structure from `require`? why return true if tx could not be sent?

Reply:
Coz I don't want the test run to terminate if it fails to send a transaction. There's a check for the failure late on.

Reply:
where is this check and how do you know there was a problem if you return no problem?